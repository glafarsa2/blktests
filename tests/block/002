#!/bin/bash
#
# Test debugfs interaction with blktrace. Removes a device while blktrace is
# still running.
#
# Copyright (C) 2017 Omar Sandoval
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

. common/scsi_debug

DESCRIPTION="remove a device while running blktrace"
QUICK=1

requires() {
	_have_blktrace && _have_scsi_debug
}

test() {
	echo "Running ${TEST_NAME}"

	if ! _get_scsi_debug_dev; then
		return 1
	fi

	blktrace -D "$TMP_DIR" "$SCSI_DEBUG_DEV" >"$FULL" 2>&1 &
	sleep 0.5
	echo 1 > "/sys/block/${SCSI_DEBUG_NAME}/device/delete"
	if [[ ! -d /sys/kernel/debug/block/${SCSI_DEBUG_NAME} ]]; then
		echo "debugfs directory deleted with blktrace active"
	fi
	{ kill $!; wait; } >/dev/null 2>/dev/null
	if [[ -d /sys/kernel/debug/block/${SCSI_DEBUG_NAME} ]]; then
		echo "debugfs directory leaked"
	fi

	_put_scsi_debug_dev

	echo "Test complete"
}
