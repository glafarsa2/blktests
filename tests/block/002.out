Running block/002
Test complete
